File: java/Solution704.java
package leetCode.java;

//Binary Search O(logn) O(1)
class Solution704 {
    public int search(int[] nums, int target) {
        //nums sorted in ascending order
        int l = 0, r = nums.length - 1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else { // num[mid] > target
                r = mid - 1;
            }
        }
        return -1;
    }
}

File: c++/Solution704.cpp
//Binary Search O(logn) O(1)
class Solution704 {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while(l <= r){
            int mid = (l + r) / 2;
            if(target == nums[mid]) return mid;
            if(target < nums[mid]){
                r = mid - 1;
            }else{
                l = mid + 1;
            }
        }
        return -1;
    }
};